model-server-openapi/specifications/model-server.yaml

```diff
@@ -13,481 +13,700 @@ info:
+      - $ref: "#/components/parameters/BranchPath"
+
+    post:
+      operationId: postRepositoryBranchQuery
+      tags:
+        - ModelQl access
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
       responses:
-        "403":
-          $ref: '#/components/responses/403'
-        "401":
-          $ref: '#/components/responses/401'
-        "500":
-          $ref: '#/components/responses/500'
         "200":
-          $ref: '#/components/responses/versionDelta'
-  /v2/repositories/{repository}/branches/{branch}/pollHash:
+          $ref: '#/components/responses/200json'
+        default:
+          $ref: '#/components/responses/Problem'
+
+  /repositories/{repository}/versions/{versionHash}/delta:
+    parameters:
+      - $ref: "#/components/parameters/RepositoryPath"
+      - $ref: "#/components/parameters/VersionHashPath"
     get:
-      operationId: pollRepositoryBranchHash
+      operationId: getRepositoryVersionHash
+      description: |-
+        If `from` is not specified, this **returns the full version data**.
+      tags:
+        - versions
+        - delta access
       parameters:
-        - name: lastKnown
-          in: "query"
-          required: false
-          schema:
-            type: string
-        - name: repository
-          in: "path"
-          required: true
-          schema:
-            type: string
-        - name: branch
-          in: "path"
-          required: true
-          schema:
-            type: string
+        - $ref: "#/components/parameters/FromHashQuery"
       responses:
-        "403":
-          $ref: '#/components/responses/403'
-        "401":
-          $ref: '#/components/responses/401'
-        "500":
-          $ref: '#/components/responses/500'
         "200":
-          $ref: '#/components/responses/200'
-  /v2/repositories/{repository}/branches/{branch}/query:
+          $ref: "#/components/responses/VersionDelta"
+        "404":
+          description: Repository or version hash does not exist.
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/Problem"
+        default:
+          $ref: "#/components/responses/Problem"
+
+  # TODO specify input and output formats
+  /repositories/{repository}/versions/{versionHash}/query:
+    parameters:
+      - $ref: "#/components/parameters/RepositoryPath"
+      - $ref: "#/components/parameters/VersionHashPath"
+
     post:
-      operationId: postRepositoryBranchQuery
-      parameters:
-        - name: repository
-          in: "path"
-          required: true
-          schema:
-            type: string
-        - name: branch
-          in: "path"
-          required: true
-          schema:
-            type: string
+      operationId: queryRepositoryVersionHash
+      tags:
+        - ModelQl access
+        - versions
```

Comment:
`/repositories/{repository}/versions/{versionHash}/query` does not need `versions`. `ModelQL` is enough.

Reply:
See above